File: 백준/Silver/2003. 수들의 합 2/수들의 합 2.py
import sys

input = sys.stdin.readline


n, m = map(int, input().split())
numbers = [0] + list(map(int, input().split()))
prefixSum = [0] * (n + 1)
for i in range(1, n + 1):
    prefixSum[i] = prefixSum[i - 1] + numbers[i]

answer = 0
start = 1
end = 1
while end < n + 1:
    s = prefixSum[end] - prefixSum[start - 1]
    if s == m:
        start += 1
        end += 1
        answer += 1
    elif s < m:
        end += 1
    else:
        start += 1

print(answer)

File: 백준/Silver/2003. 수들의 합 2/수들의 합 2.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.StringTokenizer;

public class Main {

  static StringBuilder sb = new StringBuilder();
  static int n, m;
  static int[] prefixSum;

  public static void main(String[] args) throws IOException {
    setUp();

    int start = 0;
    int end = 1;
    int answer = 0;
    while (end < n + 1) {
      int s = prefixSum[end] - prefixSum[start];
      if (s == m) {
        answer++;
        start++;
      } else if (s < m) {
        end++;
      } else {
        start++;
      }
    }

    sb.append(answer);

    output();
  }

  private static void setUp() throws IOException {
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringTokenizer st = new StringTokenizer(br.readLine());
    n = Integer.parseInt(st.nextToken());
    m = Integer.parseInt(st.nextToken());
    prefixSum = new int[n + 1];
    st = new StringTokenizer(br.readLine());
    for (int i = 1; i < n + 1; i++) {
      prefixSum[i] = prefixSum[i - 1] + Integer.parseInt(st.nextToken());
    }
  }

  private static void output() throws IOException {
    BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
    bw.write(sb.toString());
    bw.flush();
    bw.close();
  }

}